### DISCUSSION TITLE
Can't set tracking uri to s3 bucket

### DISCUSSION THREAD

**Discussion by Jason-F** [mlflow/mlflow#5866] [Q&A] [answered]
I'm trying to set the tracking uri to an s3 bucket. I don't want to run a server but just store that info on the s3 bucket.

If I set the tracking uri to local eg. "data/mlflow" it will store that info no propbelm but if I try store it to : "s3://bucket_location/mlflow" it gives me this error:

"mlflow.tracking.registry.UnsupportedModelRegistryStoreURIException:  Model registry functionality is unavailable; got unsupported URI 's3://bucket_location/mlflow/' for model registry data storage. Supported URI schemes are:['', 'file', 'databricks', 'http', 'https', 'postgresql', 'mysql', 'sqlite', 'mssql']. See https://www.mlflow.org/docs/latest/tracking.html#storage for how to run an MLflow server against one of the supported backend storage locations"

Any help around this? I wasn't able to find a solution in that errors link

I can store my artifacts on that s3 location
here is some of the code:
```
    mlflow.set_tracking_uri(mlflow_tracking_uri) #this sets the tracking uri but won't store the info on the s3 bucket...only works on local
    mlflow.create_experiment( mlflow_experiment_name, s3_bucket) #this stores the artifacts on the s3 bucket
```

**Comment by BenWilson2** [comment]
The tracking uri is for the tracking server (i.e., an RDBMS) that stores the metrics, parameters, tags, etc. for runs and experiments. Artifacts are, as you inferred above, stored on a file store. You can't write a model or any other artifact to the tracking server. What DOES get written to the tracking server regarding artifacts is metadata associated with where that artifact is logged to (i.e. on local file system, or, in the case that you're looking for, s3). 
The tracking server can be used to proxy writes to a file store location, but even in this mode, the files are not permanently stored on the tracking server's tables (only the reference to where the files have been written). 
Please read through these 6 modes of operation and the associated graphics to get a solid understanding of how it works: 
https://www.mlflow.org/docs/latest/tracking.html#how-runs-and-artifacts-are-recorded

**Reply by Jason-F** [reply]
Thank you so much for your reply!

How do I store the metrics and parameters on an s3 bucket without a server? When I append the mlflow.create_experiment with the s3bucket location, it will only store the artifact but not the metrics and params. I wasn't able to find the answer in that link you showed me that does not involve setting up a server

**Comment by BenWilson2** [comment, marked as answer]
You can't store the metrics and params to the artifact store location (well, that isn't entirely true; you can log them as artifacts with `mlflow.log_dict()` or `mlflow.log_text()` but those won't show up in the table view of the run view in the UI (which is what I'm sure you're looking for). Those are stored in a database table (or a local directory, if you're running in local mode). In local operation mode, we set up a FileStore instance that emulates a database table's access so that the UI can interface with locally stored files to populate the UI's metrics and parameters views (Scenario 1 in the link above). In all other instances, the tracking server is interfacing with a Database to retrieve the logged metrics and parameters for a given run.
